### DISCUSSION TITLE
FYI, a developer started working on adding native support for Espressif's Zigbee SDK for ESPHome on ESP32-H2/ESP32-C6.

### DISCUSSION THREAD

**Discussion by Hedda** [ffenix113/zigbee_home#38] [General]
@ffenix113 just a heads-up since are synergies between this project and [ESPHome](https://esphome.io/) now that there is an official Zigbee SDK for ESP32.

FYI; a developer have started working on adding native support for [Espressif's Zigbee SDK](https://github.com/espressif/esp-zigbee-sdk) for ESPHome on ESP32-H2/ESP32-C6:

https://github.com/luar123/esphome_zb_sensor

There is more roadmap discussion about that and related here:

https://github.com/luar123/esphome_zb_sensor/issues/28

He also posted a comment about this in this existing feature request for ESPHome -> https://github.com/esphome/feature-requests/issues/1397#issuecomment-2019064013

_Originally posted by @luar123 in https://github.com/esphome/feature-requests/issues/1397#issuecomment-2019064013_ 
> **_I have created a simple esphome example of a zigbee sensor:_** 
>
> **_https://github.com/luar123/esphome_zb_sensor_**
> 
> _**The zigbee part is currently hard-coded and included as a custom component including a task. Changing this to an external component with more configuration possibilities would be the next step.**_
> 

For reference; as mentioned in another discussion, Espressif's Zigbee stack is also based on the same ZBOSS Open Initiative (ZOI) Zigbee stack from DRS that Nordic Semiconductor is also basing their Zigbee stack for nRF Connect SDK on, but they both built their on API on top of that, however one would suspect that the likeness would be very similar since thay are both basing their SDKs and APIs on same codebase, even though Espressif is using their own [ESP-IDF](https://github.com/espressif/esp-idf) as the base for their SDK?

https://github.com/espressif/esp-zigbee-sdk

https://docs.espressif.com/projects/esp-zigbee-sdk/en/latest/esp32/

https://en.dsr-corporation.com/news/espressif-joins-zoi

https://dsr-zoi.com/

https://dsr-iot.com/solutions/zboss/

![image](https://github.com/ffenix113/zigbee_home/assets/6320001/ba51ee8c-6b81-4247-8c9a-e2692f93a959)

Verses ZBOSS in Nordic Semi's nRF Connect SDK 

https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/protocols/zigbee/index.html 

https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/protocols/zigbee/architectures.html

![image](https://github.com/ffenix113/zigbee_home/assets/6320001/de458398-f54e-43ee-97bf-108d9ec5c7c8)

**Comment by Hedda** [comment]
> @ffenix113 just a heads-up since are synergies between this project and [ESPHome](https://esphome.io/) now that there is an official Zigbee SDK for ESP32.
> 
> FYI; a developer have started working on adding native support for [Espressif's Zigbee SDK](https://github.com/espressif/esp-zigbee-sdk) for ESPHome on ESP32-H2/ESP32-C6:
> 
> https://github.com/luar123/esphome_zb_sensor

Still off-topic but FYI; @luar123 posted a project comment that he plans on instead working on an external Zigbee component for ESPHome.

**Comment by Hedda** [comment]
FYI, upcoming ESP32-H4 and ESP32-C5 MCU SoC models announced with 802.15.4 radio for Thread and/or Zigbee (+ Bluetooth).

* The existing ESP32-H2 is a single-core CPU model with 802.15.4 radio and no WiFi.
* The new ESP32-H4 is a dual-core CPU model with 802.15.4 radio and no WiFi, plus optional support for PSRAM.
* The new  ESP32-C5 is a single-core CPU model with 802.15.4 radio and WiFi.
* The existing ESP32-C6 is a dual-core CPU model with 802.15.4 radio and WiFi , plus optional support for PSRAM.

![image](https://github.com/esphome/feature-requests/assets/6320001/06ae41fa-7347-4464-ac97-90511c43c5ae)

As I understand, these should in the future eventually all be fully supported by the ESP-IDF framework as well as the ESP-Zigbee-SDK and ESP-Matter-SDK to add more flexibility in choice of model options.

PS: Also announced is the new [ESP32-P4](https://www.cnx-software.com/2023/01/06/espressif-esp32-p4-a-400-mhz-general-purpose-dual-core-risc-v-microcontroller/) which a very powerful high-performance MCU SoC without any radios at all, so it is designed to act as a general purpose host MCU that in combination with above SoC to make a 2-SoC solution with Ethernet PHY (like for example a [Zigbee Gateway](https://docs.espressif.com/projects/esp-thread-br/en/latest/hardware_platforms.html)).

https://www.cnx-software.com/2022/09/13/esp-hosted-helps-adds-wifi-connectivity-to-legacy-linux-or-mcu-based-products/

https://docs.espressif.com/projects/esp-at/en/latest/esp32/Get_Started/What_is_ESP-AT.html

**Comment by Hedda** [comment]
@ffenix113 with the announcement of "Open Home Foundation" you might want to reconsider porting this project to ESPHome instead(?):

https://building.open-home.io/announcing-the-open-home-foundation/

* (short version at https://www.openhomefoundation.org/blog/announcing-the-open-home-foundation/)

If ported to ESPHome then you would surely get more interest from likeminded developers interested in free and open source firmware.

Summery;  ESPHome and Home Assistant are owned by this new non-profit organization called "Open Home Foundation" that has been founded by a few of the Home Assistant founders and ESPHome developers, shere are now quite a few more strong arguments for why ESPHome has the potential of becoming the go-to MCU firmware framework for free and open-source device development in the future.

I highly recommend that you watch the whole "State of the Open Home 2024" stream:

https://www.youtube.com/watch?v=oa__fLArsFk

...or at the very least least listen to what the said about their roadmap for ESPHome:

https://www.youtube.com/watch?v=oa__fLArsFk&t=8074s

They also mention other things around ESPHome several times during that video and thier requests for others to join in the develepment.

https://www.openhomefoundation.org/projects/

https://github.com/openhomefoundation

The Verge has done an interview with them on the "Open Home Foundation" and posted an article about their overall ambitions and goals:

https://www.theverge.com/24135207/home-assistant-announces-open-home-foundation

**Reply by ffenix113** [reply]
Hello Hedda,

Thanks for pointing this out. I appreciate your concern, information and updates, but for now I would like to keep this as a "private" project, and later, when this either will get some traction or when I(or community) would feel like the project is ready - we will see how to best cooperate with "Open Home Foundation".

At this point of time I would like to focus on basic things still, and it is too early to move in some other direction(at least in my opinion).

**Comment by Hedda** [comment]
@ffenix113 yet another FYI update this another off-topic project as @luar123 has created a new separate Espressif ESP32 Zigbee custom component project for ESPHome:

https://github.com/luar123/zigbee_esphome

suggest at least check out the YAML syntax if nothing else:

https://github.com/luar123/zigbee_esphome/blob/master/README.md

**Reply by ffenix113** [reply]
Thank you. 

I quickly checked it out and the configuration does seem reasonable for the thing it wants to do.

Though I still don't fully understand the purpose of the project.. You have a ESPHome board, which uses WIFI to control it, then you use that project to add Zigbee capabilities on top of it. But you need to connect the board to the Zigbee coordinator, at which points one can assume that the Zigbee network is already being used in some way, and probably also have some automations on top of it.

Sure, people can have Zigbee network not included into Home Assistant, but if it is - I don't understand what the benefit of the project is if you can do all the things through Home Assistant automations, which would be more portable solution.

I probably miss the point of the project, and would be glad if you can explain more in which case it would be useful.

**Reply by Hedda** [reply]
> You have a ESPHome board, which uses WIFI to control it, then you use that project to add Zigbee capabilities on top of it. But you need to connect the board to the Zigbee coordinator, at which points one can assume that the Zigbee network is already being used in some way, and probably also have some automations on top of it.

No, the point is to use only Zigbee with ESPHome, not WiFi or Ethernet. You can already flash via USB/serial and use ESPHome stand-alone without any connection too

**Reply by ffenix113** [reply]
Oh, I understand now.

So this will create a board, that will be able to execute automations of its own, or just behave like a sensor(s).

Then it does make sense to me, thank you.

Hope that dev will finish his project and it will be introduced in ESPHome! Though, for now at least, only some ESP boards will support this feature.

What `zigbee_home` can take from EspHome is "components", which will be referenced from somewhere, though it is a bit early for that. When we will support C++ - then it should be easier to add.

**Reply by Hedda** [reply]
> So this will create a board, that will be able to execute automations of its own, or just behave like a sensor(s).

@ffenix113 Yes both will be possible with ESPHome. The idea is that you use ESPHome support for pheripials and YAML scripting to create either just sensors or full-on stand-alone solutions with on-device automations so that it can works as any type of Zigbee device, either as a battery-powered Zigbee End Devices using ESP32-H2 for sleepy sevices or as a mains-powered Zigbee Router devices using ESP32-C6, (ESP32-H4 and ESP32-H5 are another two new ESP32 SoCs coming with Zigbee radio support too).

So users have the option to either create a very basic sensor device or a very complexes product with onboard automations.

Then after flashing that firmware you built you can pair it too any Zigbee gateway, not just Home Assistent’s ZHA integration but also others lile Zigbee2MQTT and deCONZ, as well as OpenHAB and Domoticz.
